static-files/public/safary-0.1.16.js

```diff
@@ -0,0 +1,2850 @@
+                              void 0 !== e[0] &&
+                              n.push({ address: e[0], event: 'eth_accounts__eth_listener', type: e[1], chainId: e[2] })
+                          }),
+                          t.sendWalletData(n),
+                          [2]
+                        )
+                    }
+                  })
+                })),
+                function (t) {
+                  return e.apply(this, arguments)
+                }),
+              )
+              const n = this
+              window.ethereum.on(
+                'chainChanged',
+                (function () {
+                  const e = i(function (e) {
+                    let t
+                    return u(this, function (r) {
+                      return (
+                        (t = new Array()),
+                        Array.isArray(e) || (e = [e]),
+                        e.forEach(
+                          (function () {
+                            const e = i(function (e) {
+                              return u(this, function (n) {
+                                return (
+                                  t.push({ address: 'Unknown', event: 'chainChanged', type: null, chainId: e }), [2]
+                                )
+                              })
+                            })
+                            return function (t) {
+                              return e.apply(this, arguments)
+                            }
+                          })(),
+                        ),
+                        n.sendWalletData(t),
+                        [2]
+                      )
+                    })
+                  })
+                  return function (t) {
+                    return e.apply(this, arguments)
+                  }
+                })(),
+              )
+            } else window.__defineSetter__('ethereum', this.setupEthereumListeners)
+          },
+        },
+        {
+          key: 'setupPhantomSolanaListeners',
+          value: function () {
+            if (void 0 !== window.phantom) {
+              let e,
+                t = this
+              window.phantom.solana.on(
+                'connect',
+                ((e = i(function (e) {
+                  let n
+                  return u(this, function (r) {
+                    return (
+                      (n = new Array()).push({
+                        address: e.toString(),
+                        event: 'accountsChanged',
+                        type: 'Phantom',
+                        chainId: 'Solana',
+                      }),
+                      t.sendWalletData(n),
+                      [2]
+                    )
+                  })
+                })),
+                function (t) {
+                  return e.apply(this, arguments)
+                }),
+              )
+            } else window.__defineSetter__('phantom', this.setupPhantomSolanaListeners)
+          },
+        },
+        {
```

Comment:
solana, aptos atomscan, it seems this has a lot of code for chains we are not present in. Maybe we don't mind, just curious.